### PR TITLE
Fix issue that local mode can't work properly

### PR DESCRIPTION
**Pull request by chenchengshao**
1). Add new launch file "start_simulation_server_network.launch" to support network mode.
2). Change "start_simulation_server.launch" back to original setting.

### CODE REVIEW

**Comment by jrgnicho** [comment]
Can you verify that these changes don't break the remote demo?

**Comment by chenchengshao** [comment]
Jorge:

  I'm so sorry that I forgot this item.

  I just tried today, and there's one problem when I execute the
blending(use AB robot Studio)

  The problem is: In local mode, when executing, when robot moves to the
surface and  start blending, ABB robot stops,and it reports:
            40165: Reference error
             Description
             Task T_ROB1: Reference to  unknown entire data do_PIO_8.
             Actions
             No data (or other object) with the  specified name is visible
from this program position.

 Then I copied code from "https://github.com/ros-industrial-consortium/godel",
local mode still has same issue.

  I need a little more time to address this issue.

Thanks

Chencheng.

**Comment by jrgnicho** [comment]
@chenchengshao , did you restore the system launch file(s) to its original state? Do you recall what system files were changed? We should make this a priority because at this point we aren't able to run our blending demo locally.  Thanks for your hard work.

**Comment by chenchengshao** [comment]
Jorge.

Understood, I'm working at this right now, should be fixed late today or
tomorrow.

**Comment by chenchengshao** [comment]
Jorge.

  The problem I mentioned  yesterday about the ABB robotStudio stop moving
during the execution.

  This is the solution to fix this:
  godel_robots/abb/abb_file_suite/rapid_generator/src/rapid_emitter.cpp
after deleting line 144.
   os << "SETDO " << params.output_name << ", " << value << ";\n";
   It worked fine.  ( I got this solution from Reza several months ago )

  This problem also exists in ros-industrial-consortium/godel,and I forgot
about this issue, so it confused me a lot.

  In short: the changed in current PR is working fine, but there will be
another problem when executing blending function using ABB, it can be fixed
by modifying the rapid_emitter.cpp.

  Let me know what you think.

Thanks
Chencheng.

**Comment by jrgnicho** [comment]
@chenchengshao I believe this line corresponds to the blending tool IO.  The ABB simulator doesn't have the same IO channels as the real robot.  I wonder if the RobotStudio Simulator can be setup to create a fake IO channel so that we can run the same code as with the real robot.  @Jmeyer1292, do you have any suggestions?

By the way, these changes only seem to affect the simulated rviz robot.  How do these changes fix the local mode issue?

**Comment by chenchengshao** [comment]
Jorge:

  I think The current issue with local mode is that simulated rviz robot
can't show on rviz, because I modified launch file
start_simulation_server.launch.

  After changed start_simulation_server.launch back to original setting, I
already verified everything is working except the blending tool IO issue
which we just talking about.

Thanks
Chencheng.

**Comment by jrgnicho** [comment]
@chenchengshao I'll speak with @Jmeyer1292 to see if we can resolve the IO issue on the RobotStudio side.  Was the system launch file reverted back to its original version?

**Comment by chenchengshao** [comment]
Yes, It reverted back to original version, and for network mode, I created
a new launch file.

**Comment by jrgnicho** [comment]
Do you recall which system file it was?

**Comment by chenchengshao** [comment]
For Local mode problem, the launch file is start_simulation_server.launch.

**Comment by jrgnicho** [comment]
What I recall is that there was a system level launch file (not in the catkin workspace) that had been modified and because of that we could not run the local demo.  The _start_simulation_server.launch_ is a file that resides in the catkin workspace but not in the machine system directory.

**Comment by chenchengshao** [comment]
Jorge;

sorry for the confusing, I don't think I change any system level launch
file before.

But I just found one launch file has been changed by adding machine tag.
/opt/ros/indigo/share/industrial_robot_simulator/launch/robot_interface_simulator.launch

I know Reza worked at machine tag to support GENI function before., also
copied reza in loop.

Reza:
Jorge can't run local mode in their ubuntu machine, and looks like one
launch file robot_interface_simulator.launch has modified to support GENI
function,not sure his issue comes from this file or not, if you can revert
it back and create a new one for your GENI function?

**Comment by jrgnicho** [comment]
@chenchengshao Yes that's the file I was referring to, I already cc'ed Reza on this.  Thank you

**Comment by gr-rahimi** [comment]
Hi all,

I changed this file for my experiments using GENI. Sorry for any
inconvinience.
I will fix this in a few minutes.

Thanks,
Reza

**Comment by jrgnicho** [comment]
@gr-rahimi for all future work, all code and configuration file changes should be made on repository files unless there are some special circumstances that prevents us from going that route.  Thank you for the expedient response.

**Comment by gr-rahimi** [comment]
Sure.

I dropped the changes and it is now in its initial state.

Thanks,
Reza

**Comment by jrgnicho** [comment]
@gr-rahimi thank you.
@chenchengshao do you think your fixes in this PR are needed to run the demo locally?

**Comment by chenchengshao** [comment]
Jorge, Yes, this PR will be needed for local mode also.

**Comment by jrgnicho** [comment]
OK, then I'll review it and test it on the real robot by following the local mode instructions in the _README_
